pkg/tfbridge/names.go

```diff
@@ -49,19 +49,28 @@ func PulumiToTerraformName(name string, tfs map[string]*schema.Schema) string {
 		if ok && sch.MaxItems != 1 && (sch.Type == schema.TypeList || sch.Type == schema.TypeSet) {
 			result = singularResult
 		}
+
+		if ps != nil {
+			p, ok := ps[singularResult]
+			if ok && p != nil && p.MaxItemsOne != nil && !*p.MaxItemsOne &&
+				(sch.Type == schema.TypeList || sch.Type == schema.TypeSet) {
+				result = singularResult
+			}
+		}
 	}
 	return result
 }
 
 // TerraformToPulumiName performs a standard transformation on the given name string, from Terraform's underscore_casing
 // to Pulumi's PascalCasing (if upper is true) or camelCasing (if upper is false).
-func TerraformToPulumiName(name string, sch *schema.Schema, upper bool) string {
+func TerraformToPulumiName(name string, sch *schema.Schema, ps *SchemaInfo, upper bool) string {
 	var result string
 	var nextCap bool
 	var prev rune
 
 	// Pluralize names that will become array-shaped Pulumi values
-	if sch != nil && sch.MaxItems != 1 && (sch.Type == schema.TypeList || sch.Type == schema.TypeSet) {
+	if (ps == nil || ps != nil && ps.MaxItemsOne == nil) && sch != nil && sch.MaxItems != 1 &&
```

Comment:
Is this right? It looks different than the test above?

Also might make sense to do the same thing and compute `isMaxItemsOne` separately here as well just for readability.